Modules/_interpqueuesmodule.c

```diff
@@ -1810,49 +1790,42 @@ queuesmod_is_full(PyObject *self, PyObject *args, PyObject *kwds)
     Py_RETURN_FALSE;
 }
 
-PyDoc_STRVAR(queuesmod_is_full_doc,
-"is_full(qid)\n\
-\n\
-Return true if the queue has a maxsize and has reached it.");
+/*[clinic input]
+_interpqueues.get_count
+    qid: qidarg
+
+Return the number of items in the queue.
+[clinic start generated code]*/
 
 static PyObject *
-queuesmod_get_count(PyObject *self, PyObject *args, PyObject *kwds)
+_interpqueues_get_count_impl(PyObject *module, int64_t qid)
+/*[clinic end generated code: output=fb9e66e829cdd964 input=ce47690e7598884b]*/
 {
-    static char *kwlist[] = {"qid", NULL};
-    qidarg_converter_data qidarg = {0};
-    if (!PyArg_ParseTupleAndKeywords(args, kwds,
-                                     "O&:get_count", kwlist,
-                                     qidarg_converter, &qidarg)) {
-        return NULL;
-    }
-    int64_t qid = qidarg.id;
-
     Py_ssize_t count = -1;
     int err = queue_get_count(&_globals.queues, qid, &count);
-    if (handle_queue_error(err, self, qid)) {
+    if (handle_queue_error(err, module, qid)) {
         return NULL;
     }
     assert(count >= 0);
     return PyLong_FromSsize_t(count);
 }
 
-PyDoc_STRVAR(queuesmod_get_count_doc,
-"get_count(qid)\n\
-\n\
-Return the number of items in the queue.");
+/*[clinic input]
+_interpqueues._register_heap_types
+    queuetype: object
+    emptyerror: object
+    fullerror: object
```

Comment:
We can use `subclass_of=` for type checking in Argument Clinic.

Reply:
Does `subclass_of="&BaseException_Type"` not work?

Reply:
The exported name is `PyObject *PyExc_BaseException`, `PyTypeObject _PyExc_BaseException` is static. So we would need to do `object(subclass_of='(PyTypeObject *)&PyExc_BaseException')`, which feels too hacky to me. What do you think?

Reply:
Ah, sorry, this does not work. This specifies the type of the argument, but all exception _types_ have type `type`.